viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None or genre == None or rating == None:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
 
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for ele in user_data["watchlist"]:  # ele is a dict
+        if ele["title"] == title:
+            user_data["watchlist"].remove(ele)
+            user_data["watched"].append(ele)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0
+    else:
+        total = 0
+        for ele in user_data["watched"]:
+            total += ele["rating"]
+        return total/len(user_data["watched"])
+    # retun round( total/len(user_data["watched"]), 1)
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"] == []:
+        return None
+    my_dic = {}
+    max = 0
+    result = ""
+    for ele in user_data["watched"]:
+        if ele["genre"] in my_dic:
+            my_dic[ele["genre"]] += 1
+        else:
+            my_dic[ele["genre"]] = 1
+    for k, v in my_dic.items():
+        if v > max:
+            max = v
+            result = k
+    return result
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    mine = user_data["watched"]
+    if not mine:
+        return []
+    result = []

```

Comment:
If we initialized `result` in the line prior to the if-statement when we could return `result` instead of a new empty list:

```suggestion
    result = []
    if not mine:
        return result
```